engine/src/main/java/org/destinationsol/game/projectile/Projectile.java

```diff
@@ -115,25 +119,64 @@ private DSParticleEmitter buildEffect(SolGame game, EffectConfig ec, DrawableLev
     @Override
     public void update(SolGame game) {
         body.update(game);
+
         if (obstacle != null) {
-            if (!wasDamageDealt) {
-                if (config.aoeRadius >= 0) { //If AoE is enabled for this Projectile, damage all within the radius.
-                    game.getObjectManager().doToAllCloserThan(config.aoeRadius, this, (SolObject obj) ->
-                            obj.receiveDmg(config.dmg, game, body.getPosition(), config.dmgType)
-                    );
-                } else {
-                    obstacle.receiveDmg(config.dmg, game, body.getPosition(), config.dmgType);
+
+            //This handles collision of a Projectile with an entity.
+            if (obstacle instanceof SolObjectEntityWrapper) {
+                EntityRef entity = ((SolObjectEntityWrapper) obstacle).getEntity();
+
+                if (!wasDamageDealt) {
+                    if (config.aoeRadius >= 0) { //This checks if the projectile does Area-Of-Effect damage. If it does not, the value is usually -1
+                        EntityIterator iterator = game.getEntitySystemManager().getEntityManager().iterate(new Position());
+                        while (iterator.next()) {
+                            Vector2 entityPosition = iterator.getEntity().getComponent(Position.class).get().position;
+                            if (getPosition().dst2(entityPosition) <= config.aoeRadius) {
+                                game.getEntitySystemManager().sendEvent(new DamageEvent(config.dmg), entity);
+                            }
+                        }
+
+                    } else {
+                        game.getEntitySystemManager().sendEvent(new DamageEvent(config.dmg), entity);
+                    }
                 }
-            }
-            if (config.density > 0) {
-                obstacle = null;
-                wasDamageDealt = true;
-            } else {
-                collided(game);
-                if (config.emTime > 0 && obstacle instanceof SolShip) {
-                    ((SolShip) obstacle).disableControls(config.emTime, game);
+
+                if (config.density > 0) {
+                    obstacle = null;
+                    wasDamageDealt = true;
                 }
+
+                collided(game);
+
+                //TODO Once SolShip is an entity, this should be refactored to work with it
```

Comment:
Have you got a document somewhere listing all of the TODOs that should become TODONEs by the end of the project? It would be really helpful if you could share that